pkg/libvirttools/event_handler.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2020 Authors of Arktos
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package libvirttools
+
+import (
+	"github.com/Mirantis/virtlet/pkg/metadata"
+	"github.com/Mirantis/virtlet/pkg/metadata/types"
+	"github.com/golang/glog"
+	"github.com/libvirt/libvirt-go"
+	"time"
+)
+
+// handle libvirt domain events
+// currently it is merely for the memory device add/remove event to avoid repeated calls of
+// UpdateContainerResources() from kubelet while a resource updating is in progress
+//
+type eventHandler struct {
+	uri       string
+	conn      *libvirt.Connect
+	metaStore metadata.ContainerStore
+}
+
+func init() {
+	libvirt.EventRegisterDefaultImpl()
+}
+
+func NewEventHandler(uri string, store metadata.Store) *eventHandler {
+	conn, err := libvirt.NewConnect(uri)
+	if err != nil {
+		glog.Errorf("failed to connect to %v", uri)
+		return nil
+	}
+
+	return &eventHandler{
+		uri:       uri,
+		conn:      conn,
+		metaStore: store,
+	}
+}
+
+// TODO: add failure handling logic in callback registration and handler logic
```

Comment:
Same, for all TODOs please file issue to track them.

Reply:
filed issue #58 to track